sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -5792,6 +5793,75 @@ protected internal virtual BlobContainerClient GetParentBlobContainerClientCore(
             return _parentBlobContainerClient;
         }
         #endregion
+        #region PartitionedDownloader
+        internal PartitionedDownloader<BlobRequestConditions, BlobDownloadStreamingResult> GetPartitionedDownloader(
+            StorageTransferOptions transferOptions,
+            string operationName = null)
+            => new PartitionedDownloader<BlobRequestConditions, BlobDownloadStreamingResult>(
+                GetPartitionedDownloaderBehaviors(this),
+                SanitizePartitionedDownloaderOptions(transferOptions),
+                operationName);
+
+        internal static PartitionedDownloader<BlobRequestConditions, BlobDownloadStreamingResult>.Behaviors GetPartitionedDownloaderBehaviors(BlobBaseClient client)
+        {
+            return new PartitionedDownloader<BlobRequestConditions, BlobDownloadStreamingResult>.Behaviors
+            {
+                SingleDownload = async (range, args, rangeGetContentHash, async, cancellationToken, etag)
+                    => await client.DownloadStreamingInternal(
+                        range,
+                        args,
+                        rangeGetContentHash,
+                        $"{nameof(BlobBaseClient)}.{nameof(DownloadTo)}",
+                        async,
+                        cancellationToken).ConfigureAwait(false),
+                ModifyConditions = (args, etag)
+                    => args?.WithIfMatch(etag) ?? new BlobRequestConditions { IfMatch = etag },
```

Comment:
Can this be moved to SingleDownload delegate, i.e. something like this:

```
            return new PartitionedDownloader<BlobRequestConditions, BlobDownloadStreamingResult>.Behaviors
            {
                SingleDownload = async (range, args, rangeGetContentHash, async, cancellationToken, etag)
                    => { 
                            args?.WithIfMatch(etag) ?? new BlobRequestConditions { IfMatch = etag } // this may need some more conditions.
                            return await client.DownloadStreamingInternal(
                            range,
                            args,
                            rangeGetContentHash,
                            $"{nameof(BlobBaseClient)}.{nameof(DownloadTo)}",
                            async,
                            cancellationToken).ConfigureAwait(false)
                        },
                Scope = operationName => client.ClientConfiguration.ClientDiagnostics.CreateScope(operationName
                        ?? $"{nameof(Azure)}.{nameof(Storage)}.{nameof(Blobs)}.{nameof(BlobBaseClient)}.{nameof(BlobBaseClient.DownloadTo)}")
            };
```